core/src/main/java/org/elasticsearch/plugins/PluginManagerCliParser.java

```diff
@@ -0,0 +1,239 @@
+
+import com.google.common.base.Strings;
+import org.apache.commons.cli.CommandLine;
+import org.elasticsearch.common.cli.CliTool;
+import org.elasticsearch.common.cli.CliToolConfig;
+import org.elasticsearch.common.cli.Terminal;
+import org.elasticsearch.common.collect.Tuple;
+import org.elasticsearch.common.logging.log4j.LogConfigurator;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.node.internal.InternalSettingsPreparer;
+import org.elasticsearch.plugins.PluginManager.OutputMode;
+
+import java.io.IOException;
+import java.util.Locale;
+
+import static org.elasticsearch.common.cli.CliToolConfig.Builder.cmd;
+import static org.elasticsearch.common.cli.CliToolConfig.Builder.option;
+import static org.elasticsearch.common.settings.Settings.EMPTY;
+
+public class PluginManagerCliParser extends CliTool {
+
+    // By default timeout is 0 which means no timeout
+    public static final TimeValue DEFAULT_TIMEOUT = TimeValue.timeValueMillis(0);
+
+    private static final CliToolConfig CONFIG = CliToolConfig.config("plugin", PluginManagerCliParser.class)
+            .cmds(ListPlugins.CMD, Install.CMD, Remove.CMD)
+            .build();
+
+    public static void main(String[] args) {
+        Tuple<Settings, Environment> initialSettings = InternalSettingsPreparer.prepareSettings(EMPTY, true, Terminal.DEFAULT);
+        LogConfigurator.configure(initialSettings.v1());
+        int status = new PluginManagerCliParser().execute(args);
+        System.exit(status);
+    }
+
+    public PluginManagerCliParser() {
+        super(CONFIG);
+    }
+
+    public PluginManagerCliParser(Terminal terminal) {
+        super(CONFIG, terminal);
+    }
+
+    @Override
+    protected Command parse(String cmdName, CommandLine cli) throws Exception {
+        switch (cmdName.toLowerCase(Locale.ROOT)) {
+            case Install.NAME:
+                return Install.parse(terminal, cli);
+            case ListPlugins.NAME:
+                return ListPlugins.parse(terminal, cli);
+            case Remove.NAME:
+                return Remove.parse(terminal, cli);
+            default:
+                assert false : "can't get here as cmd name is validated before this method is called";
+                return exitCmd(ExitStatus.USAGE);
+        }
+    }
+
+    /**
+     * List all installed plugins
+     */
+    static class ListPlugins extends CliTool.Command {
+
+        private static final String NAME = "list";
+
+        private static final CliToolConfig.Cmd CMD = cmd(NAME, ListPlugins.class).build();
+        private final OutputMode outputMode;
+
+        public static Command parse(Terminal terminal, CommandLine cli) {
+            OutputMode outputMode = OutputMode.DEFAULT;
+            if (cli.hasOption("s")) {
+                outputMode = OutputMode.SILENT;
+            }
+            if (cli.hasOption("v")) {
```

Comment:
We cant prevent the user to run with `-s -v` - where I wanted verbose to win in that case instead of throwing an error. You think that should be changed?